### PR TITLE
change vcr to not drop tests

### PR DESCRIPTION
**Pull request by ScottSuarez**
vcr, don't drop tests. pls pls

closes https://github.com/hashicorp/terraform-provider-google/issues/9791

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .ci/containers/terraform-vcr-tester/run_vcr_tests.sh [hunk 1]
```diff
@@ -111,44 +111,56 @@ NEWLINE=$'\n'
...
+    # exit 0 because this script didn't have an error; the failure
+    # is reported via the Github Status API
+    exit 0
+  fi
+
+  NEW_RUN_TESTS=$(cat tests.json | jq -r '.testOccurrence | map(select(.status != "UNKNOWN"))| map(.name) | .[]')
+  NEW_FAILED_TESTS=$(cat tests.json | jq -r '.testOccurrence | map(select(.status == "FAILURE")) | map(.name) | join("|")')
+  if [ -n "$NEW_RUN_TESTS" ] && [ -n "$RUN_TESTS" ]
+  then
+    RUN_TESTS+="|"
```
Comment (melinath) [inline .ci/containers/terraform-vcr-tester/run_vcr_tests.sh:135]: this should use `${NEWLINE}` instead; `RUN_TESTS` is a list of one test name per line.

**Review by melinath — CHANGES_REQUESTED** [review]
Gotcha, so the issue is that while tests within each loop are formatted correctly, the lists from consecutive loops are jammed together without a connecting character, which means those tests are not accounted for correctly.

LGTM apart from the one thing.

File: .ci/containers/terraform-vcr-tester/run_vcr_tests.sh (hunk 1 above)
Comment (ScottSuarez) [reply .ci/containers/terraform-vcr-tester/run_vcr_tests.sh]: I think I got it $'\n'

**Comment by ScottSuarez** [comment]
yesssir, cool, sounds good :)

**Pull request merged**